scipy/optimize/_minimize.py

```diff
@@ -803,6 +839,62 @@ def minimize_scalar(fun, bracket=None, bounds=None, args=(),
         raise ValueError('Unknown solver %s' % method)
 
 
+def _split_bounds(bounds):
+    """Splits bounds into upper and lower parts"""
+    if isinstance(bounds, Bounds):
+        lb = bounds.lb
+        ub = bounds.ub
+    else:
+        bounds_a = np.array(bounds)
+        lb = bounds_a[:, 0]
+        ub = bounds_a[:, 1]
+    return lb, ub
+
+
+def _remove_from_bounds(bounds, i_fixed):
+    """Removes bounds for which upper and lower parts are equal"""
+    if isinstance(bounds, Bounds):
+        bounds.lb = bounds.lb[~i_fixed]
+        bounds.ub = bounds.ub[~i_fixed]
+    else:
+        bounds = np.array(bounds)[~i_fixed, :].tolist()
+    return bounds
+
+
+def _remove_from_func(fun_in, i_fixed, x_fixed, min_dim=None, remove=0):
+    """Wraps a function such that fixed variables need not be passed in"""
+    def fun_out(x_in, *args, **kwargs):
+        x_out = np.zeros_like(i_fixed, dtype=x_in.dtype)
+        x_out[i_fixed] = x_fixed
+        x_out[~i_fixed] = x_in
+        y_out = fun_in(x_out, *args, **kwargs)
+        y_out = np.array(y_out)
```

Comment:
If `fun_in` is a callable, I don't think we want to do anything with the output because it's not going to be used by the optimizer.

Reply:
if a `callback` is provided to minimize then it will need to be fed unfactored `x`.

Reply:
I think this is resolved, right?

Reply:
Yes, but it needs a test (there's one in my PR).

Reply:
Merged in mdhaber/scipy#61